### ISSUE TITLE
Pictures sent via MMS flip sideways. 

### ISSUE THREAD

**Issue by Openanonwriter** [ubports/ubuntu-touch#1751]
- Device: Pixel 3a (Sargo)
- Channel: RC
- Build: 2021-W24

Pictures flip side ways when sending MMS. They show correctly in the gallery application, when sent they flip 90 degrees clockwise and and send like that. Receiver also gets then turned side ways.   

### Steps to reproduce
1. Take Picture with phone pixel 3a (Sargo)
2.  Open up messages
3. Select photo from gallery (At this point you should see it sideways)
4.  Send

### Expected behavior
Photos should stay the same orientation as in the gallery.  

### Actual behavior
Pictures flip 90 Degrees when sending. 

### Logfiles and additional information

Chat on ubports forms here. 

https://forums.ubports.com/topic/6304/picture-messages-sent-sideways?_=1623893861831

**Comment by lduboeuf** [comment]
still an issue ?

**Comment by Flohack74** [comment]
@lduboeuf I think its a display error like I had with background and camera preview, basically you need to do this on the picture to prevent rotation on devicess with rotated sensors: https://gitlab.com/ubports/development/apps/camera-app/-/merge_requests/178/diffs?commit_id=6beed8b66abeef5a446caeff455d912252f335d0

**Comment by lduboeuf** [comment]
@Openanonwriter 
if you dowbload a picture from the internet that is well displayed and send it,  is it ok ?

**Comment by lduboeuf** [comment]
@Flohack74 iirc i saw somewhere that people complained about the camera and orientation, could it be related ? I mean camera app not storing the image wirh correct information ?

**Comment by lduboeuf** [comment]
because what intrigues me is that MMS will send the raw file as it is

**Comment by Flohack74** [comment]
No the thing is: If a camera sensor is mounted sideways or even rotated by 180° etc then the picture is delivered to camera app like this. Actually camera app is not doing much with the picture. But: In the EXIF data there is a hint for rotation stored. So all image processing and display must honour the EXIF info. And this does not happen in QML by default, leading to rotated images.
The actual image on disk is fine. We would loose quality if we try to rotate it.

**Comment by lduboeuf** [comment]
@Flohack74  ok i see, so there is something to be done in every app that display an image.
But what about the image sent through MMS, there is no special processing https://gitlab.com/ubports/development/core/telephony-service/-/blob/ubports/xenial/handler/messagesendingjob.cpp#L286
The image is sent like it is.

**Comment by Flohack74** [comment]
@lduboeuf yes that is fine. The image is sent as it is, but needs to be rotated for human display ;)
